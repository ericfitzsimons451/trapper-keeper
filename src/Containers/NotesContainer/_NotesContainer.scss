.notes-container {
    width: 80%;
    margin: 0px auto;
    display: flex;
    flex-wrap: wrap;
    padding: 40px 80px;
}