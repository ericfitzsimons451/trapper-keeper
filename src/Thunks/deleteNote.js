import { deleteCard, isLoading, hasError } from "../actions";

export const deleteNote = (id) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true));
      const response = await fetch(`http://localhost:3000/api/v1/notes/${id}`, {
        method: "DELETE"
      });
      if (response.status !== 204) {
        throw new Error(response.statusText);
      }
      dispatch(isLoading(false));
      dispatch(deleteCard(id));
    } catch (error) {
      console.log("hello world")
      dispatch(hasError(error.message));
    }
  };
};
