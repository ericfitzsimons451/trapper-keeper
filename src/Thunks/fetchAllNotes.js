import { getAllNotes, hasError, isLoading } from '../actions'

export const fetchAllNotes = (url) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      if (!response.ok) {
        throw new Error(response.statusText)
      }
      const notes = await response.json()
      dispatch(isLoading(false))
      dispatch(getAllNotes(notes))
    } catch (error) {
      dispatch(hasError(error.message))
    }
  }
}