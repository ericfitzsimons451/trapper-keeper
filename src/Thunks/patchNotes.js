import { hasError } from '../actions'

export const patchNotes = (notes) => {
	const option = {
		method: "PATCH",
		headers: {
			'Content-Type': 'application/json'
		},
		body: JSON.stringify(notes)
	}

	return async (dispatch) => {
		try {
			const response = await fetch("http://localhost:3000/api/v1/notes", option)
			if (!response.ok) {
				throw new Error(response.statusText)
			}
		} catch (error) {
			dispatch(hasError(error.message))
		}
	}
}