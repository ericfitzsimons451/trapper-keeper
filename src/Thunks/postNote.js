import { addNote, isLoading, hasError } from '../actions'

export const postNote = (note) => {
    return async (dispatch) => {
        try {
            dispatch(isLoading(true))
            const response = await fetch('http://localhost:3000/api/v1/notes')
            if (!response.ok) {
                throw new Error(response.statusText)
            }
            const note = await response.json()
            dispatch(isLoading(false))
            dispatch(addNote(note))
        } catch (error) {
            dispatch(hasError(error.message))
        }
    }
}