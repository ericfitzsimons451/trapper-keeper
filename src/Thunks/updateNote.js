import { updateNote, isLoading, hasError } from '../actions'

export const editNote = (noteToUpdate) => {
  console.log('Update Note Thunk Fires')
  const option = {
    method: 'PUT',
    body: JSON.stringify(noteToUpdate),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(`http://localhost:3000/api/v1/notes/${noteToUpdate.id}`, option)
      if (!response.ok) {
        throw new Error(response.statusText)
      }
      const note = await response.json()
      dispatch(isLoading(false))
      console.log(noteToUpdate)
      dispatch(updateNote(noteToUpdate))
    } catch (error) {
      dispatch(hasError(error.message))
    }
  }
}