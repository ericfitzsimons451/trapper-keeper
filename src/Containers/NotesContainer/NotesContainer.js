import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchAllNotes } from "../../Thunks/fetchAllNotes";

export class NotesContainer extends Component {
 

  async componentDidMount() {
    const url = "http://localhost:3000/api/v1/notes";
    await fetchAllNotes(url);
  }

  render() {
    if (this.props.notes) {
      return (
        <ul>
          {this.props.notes.map((note,index ) => {
            return (
              <div key={index}>
                <li>{note.id}</li>
                <li>
                  {note.body.map((body,index) => {
                    return <div key={index}>{body.context}</div>;
                  })}
                </li>
              </div>
            );
          })}
        </ul>
      );
    } else {
      return <div>Loading</div>;
    }
  }
}

export const mapStateToProps = state => ({
  notes: state.notes,
  errorMsg: state.errorMsg,
  isLoading: state.isLoading
});

export const mapDispatchToProps = dispatch => ({
  fetchAllNotes: url => dispatch(fetchAllNotes(url))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NotesContainer);
