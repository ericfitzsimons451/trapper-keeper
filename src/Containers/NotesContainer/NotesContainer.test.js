import React from 'react'
import { NotesContainer } from './NotesContainer'
import { mapStateToProps, mapDispatchToProps } from './NotesContainer'
import { fetchAllNotes } from '../../Thunks/fetchAllNotes'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
jest.mock('../../Thunks/fetchAllNotes')

describe('NotesContainer', () => {
   
    let wrapper

    it.skip('should render', () => {
        let component = shallow(<NotesContainer />)
        const tree = toJson(component)
        expect(tree).toMatchSnapshot()
    })

    describe('mapStateToProps', () => {
        it('should mapStateToProps', () => {
            const mockState = {
                notes: [{}, {}],
                errorMsg: 'Error message',
                isLoading: false
            }
            const mappedProps = mapStateToProps(mockState)
            expect(mappedProps).toEqual(mockState)
        })
        
    })
    
    describe('mapDispatchToProps', () => {
        it('calls dispatch with the correct action', async () => {
            const mockUrl = 'www.url.com'
            const mockDispatch = jest.fn()
            const actionToDispatch = await fetchAllNotes(mockUrl)
            const mappedProps = mapDispatchToProps(mockDispatch)
            mappedProps.fetchAllNotes(mockUrl)
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
        })
    })

    describe.skip('componentDidMount', () => {
        it('should call fetchAllNotes with the correct url', async () => {
            await wrapper.instance().componentDidMount()
            expect(fetchAllNotes).toHaveBeenCalledWith("http://localhost:3000/api/v1/notes")
        })
    })
})