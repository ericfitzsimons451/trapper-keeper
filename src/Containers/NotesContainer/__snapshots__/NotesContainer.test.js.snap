// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesContainer should match the snapshot with all data passed in 1`] = `
<div
  className="notes-container"
>
  <Connect(Note)
    note={Object {}}
  />
  <Connect(Note)
    note={Object {}}
  />
</div>
`;

exports[`NotesContainer should render the Loader if there are no notes 1`] = `<Loader />`;
